tests/drivers/rtc/rtc_api/src/test_alarm.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2022 Bjarki Arge Andreasen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/timeutil.h>
+
+#include <time.h>
+
+/* Fri Jan 01 2021 13:29:50 GMT+0000 */
+#define RTC_TEST_ALARM_SET_TIME		      (1609507790)
+#define RTC_TEST_ALARM_TEST_NOT_PENDING_DELAY (3)
+#define RTC_TEST_ALARM_TEST_PENDING_DELAY     (10)
+#define RTC_TEST_ALARM_TIME_MINUTE	      (30)
+#define RTC_TEST_ALARM_TIME_HOUR	      (13)
+
+static const struct device *rtc = DEVICE_DT_GET(DT_ALIAS(rtc));
+static const uint16_t alarms_count = DT_PROP(DT_ALIAS(rtc), alarms_count);
+
+ZTEST(rtc_api, test_alarm)
```

Comment:
This test should explicitly clear alarms on exit, or use the `after_fn` to do so.